### PR TITLE
Straw-man attempt at an introduction

### PR DESCRIPTION
**Pull request by shs96c**
Address issue #360.

---
This change is [Reviewable](https://reviewable.io/reviews/w3c/webdriver/536)

### CODE REVIEW

**Comment by shs96c** [comment]
Feel free to shred this to pieces, reword and move to a different place in the spec.

File: webdriver-spec.html [hunk 1]
```diff
@@ -89,6 +90,68 @@
...
+<p>This specification attempts to follow a number of design goals:
+
+<section>
+<h4>Compatibility with existing Selenium WebDriver functionality</h4>
+
+<p>This spec is derived from the popular
+ <a href=https://seleniumhq.org>Selenium WebDriver</a> browser
+ automation framework. Selenium is a long-lived project, and due to
+ its age and breadth of use it has a wide range of expected
+ functionality. This specification uses these expectationsw to inform
```
Comment (abotalov) [inline webdriver-spec.html]: expectationsw -> expectations

File: webdriver-spec.html [hunk 2]
```diff
@@ -486,6 +487,66 @@
...
+<p><i>This section is non-normative</i>
+
+<h3>Design Notes</h3>
+
+<p>This specification attempts to follow a number of design goals:
+
+<section>
+<h4>Compatibility with existing Selenium WebDriver functionality</h4>
+
+<p>This spec is derived from the popular
```
Comment (AutomatedTester) [inline webdriver-spec.html]: s/spec/specification

File: webdriver-spec.html [hunk 3]
```diff
@@ -486,6 +487,66 @@
...
+
+<section>
+<h4>Suitability for use end-user browser automation needs</h4>
+
+<p>It is expected that the largest group of users of this
+ specification are going to be software developers and testers writing
+ automated tests and other tooling, such as monitoring or load
+ testing, that relies on automating a browser. As such, care has been
+ taken to provide commands that simplify common tasks such
+ as <a href=#element-send-keys>typing into</a>
```
Comment (AutomatedTester) [inline webdriver-spec.html]: remove the `href` and replace with `data-lt='Element Send Keys'` for send keys and `data-lt='Element Click'`

File: webdriver-spec.html [hunk 4]
```diff
@@ -486,6 +487,66 @@
...
+ and <a href=#element-click>clicking</a> elements.
+
+</section>
+
+<section>
+<h4>Use by other specifications</h4>
+
+<p>WebDriver aims to allow other specifications
+ to <a href=#protocol-extensions>extend the spec</a> to support new
```
Comment (AutomatedTester) [inline webdriver-spec.html]: use the relevant `data-lt` here

File: webdriver-spec.html [hunk 5]
```diff
@@ -486,6 +487,66 @@
...
+ functionality, and to allow conformance tests that cannot be
+ implemented entirely in Ecmascript to be written in a vendor-neutral
```
Comment (AutomatedTester) [inline webdriver-spec.html]: probably should link ecmascript

File: webdriver-spec.html (hunk 2 above)
Comment (shs96c) [inline webdriver-spec.html]: Done

File: webdriver-spec.html (hunk 3 above)
Comment (shs96c) [inline webdriver-spec.html]: Done

File: webdriver-spec.html (hunk 4 above)
Comment (shs96c) [inline webdriver-spec.html]: Done

File: webdriver-spec.html (hunk 5 above)
Comment (shs96c) [inline webdriver-spec.html]: Done

File: webdriver-spec.html [hunk 6]
```diff
@@ -1,4 +1,4 @@
-<!doctype html>
+<!Doctype html>
```
Comment (andreastt) [inline webdriver-spec.html]: Mistake edit?

File: webdriver-spec.html [hunk 7]
```diff
@@ -504,6 +505,68 @@
...
+<section>
+<h3>Design Notes</h3>
+
+<p>This specification attempts to follow a number of design goals:
+
+<section>
+<h4>Compatibility with existing Selenium WebDriver functionality</h4>
+
+<p>This specification is derived from the popular
+ <a href=https://seleniumhq.org>Selenium WebDriver</a> browser
```
Comment (andreastt) [inline webdriver-spec.html]: This URL has a TLS issue. Should link to http://www.seleniumhq.org/.

File: webdriver-spec.html [hunk 8]
```diff
@@ -504,6 +505,68 @@
...
+ automation framework. Selenium is a long-lived project, and due to
+ its age and breadth of use it has a wide range of expected
+ functionality. This specification uses these expectations to inform
+ its design. Where improvements or clarifications have been made, they
+ have been made with care to allow existing users Selenium WebDriver
+ to avoid unexpected breakages.
+</section>
+
+<section>
+<h4>Suitability for use end-user browser automation needs</h4>
```
Comment (andreastt) [inline webdriver-spec.html]: Something fishy with this title…

File: webdriver-spec.html [hunk 9]
```diff
@@ -504,6 +505,68 @@
...
+
+<p>It is expected that the largest group of users of this
+ specification are going to be software developers and testers writing
+ automated tests and other tooling, such as monitoring or load
```
Comment (andreastt) [inline webdriver-spec.html]: I would rephrase this slightly by removing the word “expected” and say that this is the _intended_ user group.

File: webdriver-spec.html [hunk 10]
```diff
@@ -504,6 +505,68 @@
...
+ taken to provide commands that simplify common tasks such
+ as <a data-lt="Element Send Keys">typing into</a>
+ and <a data-lt="Element Click">clicking</a> elements.
+
+</section>
+
+<section>
+<h4>Use by other specifications</h4>
+
+<p>WebDriver aims to allow other specifications
```
Comment (andreastt) [inline webdriver-spec.html]: s/specifications/standards/

File: webdriver-spec.html [hunk 11]
```diff
@@ -504,6 +505,68 @@
...
+ to <a data-lt="extension commands">extend the spec</a> to support new
+ functionality, and to allow conformance tests that cannot be
+ implemented entirely in
+ <a href=https://tc39.github.io/ecma262/>Ecmascript</a> to be written
```
Comment (andreastt) [inline webdriver-spec.html]: Ecmascript → ECMAScript

File: webdriver-spec.html [hunk 12]
```diff
@@ -504,6 +505,68 @@
...
+ and <a data-lt="Element Click">clicking</a> elements.
+
+</section>
+
+<section>
+<h4>Use by other specifications</h4>
+
+<p>WebDriver aims to allow other specifications
+ to <a data-lt="extension commands">extend the spec</a> to support new
+ functionality, and to allow conformance tests that cannot be
```
Comment (andreastt) [inline webdriver-spec.html]: I would rephrase this:

> WebDriver aims to allow other standards to _provide extensions_ to support new functionality, …

File: webdriver-spec.html [hunk 13]
```diff
@@ -504,6 +505,68 @@
...
+
+</section>
+
+<section>
+<h4>Compliance with other specifications</h4>
+
+<p>This specification relies as much as possible on behaviors and
+ functionality expressed in other specifications. Wherever possible,
+ this specification avoids defining new functionality unless this
+ conflicts with its other design goals.
```
Comment (andreastt) [inline webdriver-spec.html]: I think this entire paragraph can be dropped because it doesn’t say anything non-obvious.

File: webdriver-spec.html (hunk 8 above)
Comment (shs96c) [inline webdriver-spec.html]: Changed

File: webdriver-spec.html (hunk 9 above)
Comment (shs96c) [inline webdriver-spec.html]: Reworded.

File: webdriver-spec.html (hunk 10 above)
Comment (shs96c) [inline webdriver-spec.html]: Done

File: webdriver-spec.html (hunk 12 above)
Comment (shs96c) [inline webdriver-spec.html]: Done

File: webdriver-spec.html (hunk 11 above)
Comment (shs96c) [inline webdriver-spec.html]: Done

File: webdriver-spec.html (hunk 13 above)
Comment (shs96c) [inline webdriver-spec.html]: Good point. Removed.
